data/src/main/java/com/mashup/damgledamgle/repository/remote/OnboardingRepositoryImpl.kt

```diff
@@ -0,0 +1,35 @@
+package com.mashup.damgledamgle.repository.remote
+
+import com.mashup.damgledamgle.domain.entity.NickName
+import com.mashup.damgledamgle.domain.entity.base.NetworkResponse
+import com.mashup.damgledamgle.domain.repository.OnboardingRepository
+import com.mashup.damgledamgle.mapper.NickNameMapper
+import com.mashup.damgledamgle.repository.network.ServiceBuilder
+import javax.inject.Inject
+
+/**
+ *  OnboardingRepositoryImpl.kt
+ *
+ *  Created by Minji Jeong on 2022/07/25
+ *  Copyright © 2022 MashUp All rights reserved.
+ */
+
+class OnboardingRepositoryImpl @Inject constructor(
+    private val nickNameMapper: NickNameMapper
+) : OnboardingRepository {
+    override suspend fun getNickName(adjective: String?, noun: String?): NetworkResponse<NickName> {
+        val resultData = if (adjective != null) {
+            ServiceBuilder.damgleApi.getNickName(adjective = adjective)
+        } else if (noun != null) {
+            ServiceBuilder.damgleApi.getNickName(noun = noun)
+        } else {
+            ServiceBuilder.damgleApi.getNickName()
+        }
```

Comment:
```suggestion
                val resultData = when {
            adjective != null -> ServiceBuilder.damgleApi.getNickName(adjective = adjective)
            noun != null -> ServiceBuilder.damgleApi.getNickName(noun = noun)
            else -> ServiceBuilder.damgleApi.getNickName()
        }
```
개인적으로 패턴매칭을 굉장히 좋아합니다! 더 깔끔하게 느껴져요